### DISCUSSION TITLE
How to query all categories of a product in graphQL

### DISCUSSION THREAD

**Discussion by marnauortega** [swellstores/community#344] [Help] [answered]
I'm trying to get all the categories to which a product belongs with this query:

```
query getAllProducts($category: String, $amount: Int, $sort: String) {
    products(category: $category,limit: $amount,sort: $sort, page: 1) {
      results {
        id
        name
        categories {
          name
          slug
        }
      }
    }
  }
  ```
  
  But I only get the main category, not its subcategories. Is this intended? How could I achieve my goal?

**Comment by sags95** [comment]
@marnauortega Passing in `children` should be what you're looking for

```
query getAllProducts {
  products(limit: 25, page: 1, ) {
    results {
      id
      name
      slug
      price
      currency
      categories {
        slug
        topId
        children {
          name
          slug
          topId
        }
      }
      }
    }
  }
```

**Comment by marnauortega** [comment]
@sags95 Thanks a lot for your help. But I'm not sure that this does what I'm trying.
Let's say I have a category `clothes`, and two subcategories `t-shirts` and `pants`.
And I have a product that belongs to clothes > t-shirts.

As far as I can tell, this query

```
query getAllProducts {
  products(limit: 25, page: 1, ) {
    results {
      name
      categories {
        slug
        children {
          name
          slug
        }
      }
      }
    }
  }
  ```

retrieves something like this:

```
{
  "name": "Example t-shirt",
  "categories": [
    {
    "slug": "clothes",
    "children": [
        {
          "name": "T-shirts",
          "slug": "t-shirts",
        },
        {
          "name": "Pants",
          "slug": "pants",
        },
      ]
    }
  ]
}
```

So it retrieves the main category of the product and then all subcategories of this category, and then I cannot single out t-shirts. What i'd want would be something like:

```
"categories": [
    {
       "name": "Clothes",    
       "slug": "clothes",
    },
   {
       "name": "T-shirts",
       "slug": "t-shirts",
    },
]
```

Do you see what I mean? Thanks a lot!

**Reply by sags95** [reply]
@marnauortega thanks for clarifying that, there might be limitation around this with gql, let me check with the team.

**Reply by marnauortega** [reply]
Have you been able to find out what is happening? Thank you

**Reply by sags95** [reply, marked as answer]
@marnauortega It looks like by default this type of structure is not supported unfortunately. However there are a few threads here where other members have chimed in for solutions #107

**Reply by marnauortega** [reply]
Thanks, @sags95, I managed to find a workaround.

**Comment by marnauortega** [comment]
If someone could give a hint of how to achieve this, that would be very useful. Thank you!
